mdoc/Mono.Documentation/Updater/Formatters/AttributeFormatters/AttributeFormatter.cs

```diff
@@ -90,9 +90,10 @@ public virtual bool TryGetAttributeString(CustomAttribute attribute, out string
             for (int i = 0; i < attribute.ConstructorArguments.Count; ++i)
             {
                 CustomAttributeArgument argument = attribute.ConstructorArguments[i];
-                fields.Add(MakeAttributesValueString(
-                        argument.Value,
-                        argument.Type));
+                string attributesValue = MakeAttributesValueString(argument.Value, argument.Type);
+                if (Language == Consts.CppWinRt && attributesValue.StartsWith("typeof("))
```

Comment:
While this certainly works, technically ... we should avoid adding logic like "if c++ then ...". Since `CppWinRtAttributeFormatter` inherits from `AttributeFormatter`, this kind of logic should be overridden and contained in the c++ attribute formatter. That makes it easier to implement future language-specific logic, and easier to maintain.

Reply:
This can probably be accomplished by overriding `MakeAttributesValueString` and adding this logic in the c++ attribute formatter

Reply:
`MakeAttributesValueString` is a static method so it can't be overridden. I can either make it virtual and override, or override `TryGetAttributeString` that calls `MakeAttributesValueString`. I need to change `IsIgnoredAttribute` and `IgnorableAttributes` from private to protected though.

Reply:
Yeah these changes are fine ... if you need to make something non-static and virtual that's fine. This way we can better encapsulate this logic.